### PR TITLE
[Out of Date] switch CppCodeCache to new cpp_builder.

### PR DESCRIPTION
**Pull request by xuhancn**
Original PRs is damaged by confilct and rebase: https://github.com/pytorch/pytorch/pull/128303, https://github.com/pytorch/pytorch/pull/129144

This PR just switch `CppCodeCache` to new cpp_builder, it need to be validated in `fb_code` environment. 

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/cpp_builder.py [hunk 1]
```diff
@@ -677,7 +664,7 @@ def _get_openmp_args(cpp_compiler):
             fb_openmp_extra_flags = f"-Wp,-fopenmp {openmp_lib}"
             passthough_args.append(fb_openmp_extra_flags)
 
-            libs.append("omp")
+            # libs.append("omp")
```
Comment (jgong5) [inline torch/_inductor/cpp_builder.py:667]: Do we need to keep this?

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/cpp_builder.py (hunk 1 above)
Comment (xuhancn) [reply torch/_inductor/cpp_builder.py:667]: @jgong5 It seems no, according bin's log. `fb_openmp_extra_flags = f"-Wp,-fopenmp {openmp_lib}"` was setup `openmp` in `fb_code` env.

**Comment by xuhancn** [comment]
Hi @desertfire , This PR was CI passed, could you please help on import to `fb_code`?

**Comment by xuhancn** [comment]
Hi @desertfire Any feedback?

**Pull request closed without merging**